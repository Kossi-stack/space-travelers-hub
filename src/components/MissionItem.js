import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import { joinMission, leaveMission } from '../redux/missions/missions';

const MissionItem = ({ mission }) => {
  const dispatch = useDispatch();

  const handleJoinMission = (e) => {
    e.preventDefault();
    dispatch(joinMission(mission.mission_id));
  };

  const handleLeaveMission = (e) => {
    e.preventDefault();
    dispatch(leaveMission(mission.mission_id));
  };

  return (
    <tr key={mission.mission_id}>
      <td className="missions_td--bold">{mission.mission_name}</td>
      <td className="missions_td--desc">{mission.description}</td>
      <td className="align-center">
        {!mission.reserved && <span className="badge">NOT A MEMBER</span>}
        {mission.reserved && (
          <span className="badge badge-active">ACTIVE MEMBER</span>
        )}
      </td>
      <td className="align-center">
        {!mission.reserved && (
          <button type="button" className="btn-red" onClick={handleJoinMission}>
            Join Mission
          </button>
        )}
        {mission.reserved && (
          <button
            type="button"
            className="btn-gray"
            onClick={handleLeaveMission}
          >
            Leave Mission
          </button>
        )}
      </td>
    </tr>
  );
};

MissionItem.propTypes = {
  mission: PropTypes.shape({
    mission_id: PropTypes.string.isRequired,
    mission_name: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    reserved: PropTypes.bool,
  }).isRequired,
};

export default MissionItem;
