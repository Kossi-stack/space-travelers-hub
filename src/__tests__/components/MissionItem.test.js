import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../redux/configureStore';
import MissionItem from '../../components/MissionItem';

it('renders rocket item correctly', () => {
  const mission = {
    mission_id: '9D1B7E0',
    mission_name: 'Thiacom',
    description: 'some description',
    reserved: false,
  };
  const tree = renderer
    .create(
      <Provider store={store}>
        <MissionItem key={mission.mission_id} mission={mission} />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
