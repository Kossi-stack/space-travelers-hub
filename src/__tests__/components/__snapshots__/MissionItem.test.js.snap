// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders rocket item correctly 1`] = `
<tr>
  <td
    className="missions_td--bold"
  >
    Thiacom
  </td>
  <td
    className="missions_td--desc"
  >
    some description
  </td>
  <td
    className="align-center"
  >
    <span
      className="badge"
    >
      NOT A MEMBER
    </span>
  </td>
  <td
    className="align-center"
  >
    <button
      className="btn-red"
      onClick={[Function]}
      type="button"
    >
      Join Mission
    </button>
  </td>
</tr>
`;
